service/src/main.rs

```diff
@@ -285,20 +277,30 @@ fn start_worker<E, T, W, D>(
 
 	// ------------------------------------------------------------------------
 	// start worker api direct invocation server
-	println!("rpc worker server listening on ws://{}", config.worker_url());
-
-	let ws_handler_factory = Arc::new(WsHandlerFactory::new(enclave.clone(), watch_list));
-	let ws_direct_server = WsDirectServerRunner::new(ws_handler_factory, enclave.clone());
-	ws_direct_server.run(config.worker_url());
+	let direct_invocation_server_addr = config.worker_url();
+	let enclave_for_direct_invocation = enclave.clone();
+	thread::spawn(move || {
+		println!(
+			"[+] RPC direction invocation server listening on wss://{}",
+			direct_invocation_server_addr
+		);
+		enclave_for_direct_invocation
+			.init_direct_invocation_server(direct_invocation_server_addr)
+			.unwrap();
+		println!("[+] RPC direction invocation server shut down");
+	});
```

Comment:
Run the web-socket server on a separate thread inside the enclave.

Reply:
I think this is also the way we should handle block production afterwards.